### PR TITLE
feat(e2e-tests): use near sandbox worker

### PR DESCRIPTION
**Pull request by fbozic**
- setup near sandbox worker
- deploy and configure context config contract
- configure merods to use near sandbox worker
---
Run
```
========================= Starting nodes ===========================
Command: './target/debug/merod' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node1", "init", "--swarm-port", "2427", "--server-port", "2527"]
Command: './target/debug/merod' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node1", "config", "discovery.rendezvous.namespace=\"calimero/e2e-tests/3790602963\"", "context.config.new.contract_id=\"dev-20241119174216-49222027393462\"", "context.config.signer.use=\"self\"", "context.config.signer.self.near.testnet.rpc_url=\"http://127.0.0.1:50750/\"", "context.config.signer.self.near.testnet.account_id=\"node1.test.near\"", "context.config.signer.self.near.testnet.public_key=\"ed25519:Gxa24TGbJu4mqdhW3GbvLXmf4bSEyxVicrtpChDWbgga\"", "context.config.signer.self.near.testnet.secret_key=\"ed25519:3JtQnV5Tm5GM35t24mytoqR4UbLEAa6km4tiPXd6ubXebCrviQ7usSWJNKFYNyFkmtf6D2qZfN9ZUw8C2mibXw1C\""]
Command: './target/debug/merod' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node1", "run"]
Command: './target/debug/merod' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node2", "init", "--swarm-port", "2428", "--server-port", "2528"]
Command: './target/debug/merod' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node2", "config", "discovery.rendezvous.namespace=\"calimero/e2e-tests/3790602963\"", "context.config.new.contract_id=\"dev-20241119174216-49222027393462\"", "context.config.signer.use=\"self\"", "context.config.signer.self.near.testnet.rpc_url=\"http://127.0.0.1:50750/\"", "context.config.signer.self.near.testnet.account_id=\"node2.test.near\"", "context.config.signer.self.near.testnet.public_key=\"ed25519:Gxa24TGbJu4mqdhW3GbvLXmf4bSEyxVicrtpChDWbgga\"", "context.config.signer.self.near.testnet.secret_key=\"ed25519:3JtQnV5Tm5GM35t24mytoqR4UbLEAa6km4tiPXd6ubXebCrviQ7usSWJNKFYNyFkmtf6D2qZfN9ZUw8C2mibXw1C\""]
Command: './target/debug/merod' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node2", "run"]
====================================================================
================= Setting up scenario and context ==================
Loaded test scenario from file: "./e2e-tests/config/scenarios/kv-store/test.json"
TestScenario { steps: [ContextCreate(CreateContextStep { application: LocalFile("./apps/kv-store/res/kv_store.wasm") }), JsonRpcCall(JsonRpcCallStep { method_name: "set", args_json: Object {"key": String("foo"), "value": String("bar")}, expected_result_json: None, target: Inviter }), JsonRpcCall(JsonRpcCallStep { method_name: "get", args_json: Object {"key": String("foo")}, expected_result_json: Some(String("bar")), target: Inviter }), ContextInviteJoin(InviteJoinContextStep)] }
Picked inviter: node2
Picked invitees: ["node1"]
====================================================================
======================== Starting step =============================
Step: ContextCreate(CreateContextStep { application: LocalFile("./apps/kv-store/res/kv_store.wasm") })
Command: './target/debug/meroctl' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node2", "--output-format", "json", "app", "install", "--path", "./apps/kv-store/res/kv_store.wasm"]
Command: './target/debug/meroctl' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node2", "--output-format", "json", "context", "create", "-a", "Gd15bZJudKR1b9v6SrsLRUoSb4zwSMXFAYmXPXms5Myv"]
Report: Created context on 'node2' node
====================================================================
======================== Starting step =============================
Step: JsonRpcCall(JsonRpcCallStep { method_name: "set", args_json: Object {"key": String("foo"), "value": String("bar")}, expected_result_json: None, target: Inviter })
Command: './target/debug/meroctl' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node2", "--output-format", "json", "call", "H62WCCPnCgnT2xqyUc9eFBpaLt2jptdvpt8Jh5NTCAyX", "set", "--args", "{\"key\":\"foo\",\"value\":\"bar\"}", "--as", "AxwKwKxiRz1f4m1CwRF2KXXxZSFbWWcbPsJdfJWQw7sj"]
Report: Call on 'node2' node passed assertion
====================================================================
======================== Starting step =============================
Step: JsonRpcCall(JsonRpcCallStep { method_name: "get", args_json: Object {"key": String("foo")}, expected_result_json: Some(String("bar")), target: Inviter })
Command: './target/debug/meroctl' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node2", "--output-format", "json", "call", "H62WCCPnCgnT2xqyUc9eFBpaLt2jptdvpt8Jh5NTCAyX", "get", "--args", "{\"key\":\"foo\"}", "--as", "AxwKwKxiRz1f4m1CwRF2KXXxZSFbWWcbPsJdfJWQw7sj"]
Report: Call on 'node2' node passed assertion
====================================================================
======================== Starting step =============================
Step: ContextInviteJoin(InviteJoinContextStep)
Command: './target/debug/meroctl' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node1", "--output-format", "json", "identity", "generate"]
Command: './target/debug/meroctl' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node2", "--output-format", "json", "context", "invite", "H62WCCPnCgnT2xqyUc9eFBpaLt2jptdvpt8Jh5NTCAyX", "AxwKwKxiRz1f4m1CwRF2KXXxZSFbWWcbPsJdfJWQw7sj", "FkoqdbFdyiSyRRD4QWq6HdUgaceZGM1dACtRG5Sqvrb5"]
Command: './target/debug/meroctl' ["--home", "./e2e-tests/corpus/nodes", "--node-name", "node1", "--output-format", "json", "context", "join", "881rXMMvrudayPJwRM5pv9qtwbgEVR56evyatqy5G3w3", "a8YqKDkNu6WYsArMQcSCnqfeXHyRDDrSQtx5vN8vXozTTUoXnqHx3j8s9KpKHcyogqNrsR1XbDoSqcjHoq6gZksdLGZs1ajhNfdLkMx9BnWBuVHUknAFk8zPYvUtzVk6jbbfHE4sUQ9GVFJqHDVehW7mjJ2MaD2pbfVb"]
Report: Node 'node1' joined the context
====================================================================
```

### CODE REVIEW

File: e2e-tests/src/driver.rs [hunk 1]
```diff
@@ -62,16 +65,112 @@ impl Driver {
...
+        for i in 0..self.config.network.node_count {
+            let node_name = format!("node{}", i + 1);
+            let mut merods = self.merods.borrow_mut();
+            if !merods.contains_key(&node_name) {
+                let near_account = near_root_account
+                    .create_subaccount(&node_name)
+                    .initial_balance(NearToken::from_near(30))
+                    .transact()
+                    .await?
+                    .into_result()?;
```
Comment (miraclx) [inline e2e-tests/src/driver.rs]: why not use one account for this?

**Review by miraclx — APPROVED** [review]

File: e2e-tests/src/driver.rs (hunk 1 above)
Comment (fbozic) [reply e2e-tests/src/driver.rs]: I don't know, c/p from sandbox tests for that contract 😄 
You prefer one account?

Comment (miraclx) [reply e2e-tests/src/driver.rs]: yeah, it's fine to use one, effectively the same behaviour as using a relayer, but it's fine either way. lgtm

Comment (fbozic) [reply e2e-tests/src/driver.rs]: oki, will keep N in that case since it makes sense IMO (logically)

**Pull request merged**